contracts/Controller.sol

```diff
@@ -245,46 +240,41 @@ contract Controller is IController {
         // VoterProxy so that it can grab the incentive tokens off the contract after claiming rewards
         // RewardFactory so that stashes can make new extra reward contracts if a new incentive is added to the gauge
         poolInfo[pid].stash = stash;
-        IVoterProxy(staker).grantStashAccess(stash);
         IRewardFactory(rewardFactory).grantRewardStashAccess(stash);
         redirectGaugeRewards(stash, _gauge);
         emit AddedPool(pid, _lptoken, token, _gauge, newRewardPool, stash);
     }
 
-    /// @notice shuts down a currently active pool
-    /// @param _pid The id of the pool to shutdown
-    function shutdownPool(uint256 _pid) external onlyAddress(poolManager) {
-        PoolInfo storage pool = poolInfo[_pid];
+    /// @notice Shuts down multiple pools
+    /// @dev Claims rewards for that pool before shutting it down
+    /// @param _startPoolIdx Start pool index
+    /// @param _endPoolIdx End pool index (excluded)
+    function bulkPoolShutdown(uint256 _startPoolIdx, uint256 _endPoolIdx) external onlyAddress(poolManager) {
+        for (uint256 i = _startPoolIdx; i < _endPoolIdx; ++i) {
+            PoolInfo storage pool = poolInfo[i];
+
+            if (pool.shutdown) {
+                continue;
+            }
+
+            _earmarkRewards(i);
 
-        //withdraw from gauge
-        // solhint-disable-next-line
-        try IVoterProxy(staker).withdrawAll(pool.lptoken, pool.gauge) {
+            //withdraw from gauge
             // solhint-disable-next-line
-        } catch {}
+            try IVoterProxy(staker).withdrawAll(pool.lptoken, pool.gauge) {
+                // solhint-disable-next-line
+            } catch {}
```

Comment:
We can do this for an external call, so that if the case of a failure it doesn't revert, so the code will continue execution of the loop, and the rest of the code.